### PR TITLE
Stats: Modernising Subscribers on the Insights page

### PR DESCRIPTION
**Pull request by grzegorz-cp**
Related to #72134

## Proposed Changes

* modernising the `Subscribers` module behind a feature flag (Stats Insights page)
* extending Stats card with horizontal bars to display text values instead of numbers and avoid showing horizontal bars
* apply flex box wrapping to the header (only the one having a toggle) because this component is very narrow as one or 3 modules in one row
* converting an option to display an avatar from a specific module to a property 
* replacing moment.js time calculation

## Testing Instructions

* go to the live branch and the `Insights` page
* verify that the current `Subscribers` module renders correctly and select switching data between wp.com and email works correctly
* apply feature flag to the URL `?flags=stats/insights-page-grid`
* verify that the new toggle in the header switches between data sets

| Before | After |
| --- | --- |
| SCR-20230216-eif | SCR-20230216-ehc |

## Pre-merge Checklist

- [ ] Has the general commit checklist been followed? (PCYsg-hS-p2)
- [ ] [Have you written new tests](https://wpcalypso.wordpress.com/devdocs/docs/testing/index.md) for your changes?
- [ ] Have you tested the feature in Simple (P9HQHe-k8-p2), Atomic (P9HQHe-jW-p2), and self-hosted Jetpack sites (PCYsg-g6b-p2)?
- [ ] Have you checked for TypeScript, React or other console errors?
- [ ] Have you used memoizing on expensive computations? More info in [Memoizing with create-selector](https://github.com/Automattic/wp-calypso/blob/trunk/packages/state-utils/src/create-selector/README.md) and [Using memoizing selectors](https://react-redux.js.org/api/hooks#using-memoizing-selectors) and [Our Approach to Data](https://github.com/Automattic/wp-calypso/blob/trunk/docs/our-approach-to-data.md)
- [ ] Have we added the "[Status] String Freeze" label as soon as any new strings were ready for translation (p4TIVU-5Jq-p2)?
- [ ] For changes affecting Jetpack: Have we added the "[Status] Needs Privacy Updates" label if this pull request changes what data or activity we track or use (p4TIVU-ajp-p2)?

### CODE REVIEW

File: client/my-sites/stats/stats-followers/index.jsx [hunk 1]
```diff
@@ -56,18 +58,50 @@ class StatModuleFollowers extends Component {
...
+		if ( days > 0 ) {
+			result = days + ' days';
+		} else if ( hours > 0 ) {
+			result = hours + ' hours';
+		} else if ( minutes > 0 ) {
+			result = minutes + ' minutes';
+		}
+
+		return result;
 	}
```
Comment (jsnmoon) [inline client/my-sites/stats/stats-followers/index.jsx:105]: This might be good to spin off into a module outside of this function.

**Review by jsnmoon — APPROVED** [review]
This works as described; feel free to ship!

Here are some observations:

image
Clicking anywhere on the highlighted row will open the WPCOM user's main site. Could we limit this behavior to the ↗️ icon shown on hover?

Also, is the username underline intentional?

image

In the future, how should we handle empty toggled lists?

**Review by kangzj — APPROVED** [review]
It works well for me and the changes look reasonable 👍

**Comment by grzegorz-cp** [comment]
Thanks for the comments.
The underline is desired and a part of the design. However, I'm unsure about restricting the click to the redirect action icon. This will render the underline redundant and unnecessary. On one hand, the click behaviour is consistent with other horizontal bar rows, and the underline is consistent with the previous version of the Subscribers component.
We can either:
* remove the link indication and the active click area from the entire row but keep the redirect hover action, or
* keep the row click area and keep the redirect hover action (like it is in the PR), or
* remove the redirect hover action and only keep the link click action (on the whole row or only the text)

All the options are consistent with some parts of the page, so I'm unsure which to keep.
@jeffgolenski @keoshi do you have thoughts?

**Pull request merged**